docs/deploy-on-k8s.md

```diff
@@ -0,0 +1,115 @@
+
+To create the cluster (called "temporal-oms"):
+
+```sh
+./deployments/create-eks.sh
+```
+
+This will create a new EKS cluster on AWS and configure `kubectl` to talk to it.
+
+You should be able to run:
+
+```sh
+kubectl get pods
+```
+
+And see the message `No resources found in default namespace.`
+
+## Installing Temporal into Kubernetes
+
+To install a basic Temporal cluster into Kubernetes we can use the Temporal Helm Charts.
+
+To install Temporal:
+
+```sh
+./deployments/install-temporal.sh
+```
+
+This may take a while, as the Temporal cluster relies on Cassandra and Elasticsearch, which can both take a while to boot up.
+
+Once the Temporal cluster is up, the script creates the Temporal namespace "default" for you. This is Temporal namespace the OMS application expects to use, unless configured otherwise.
+
+## Installing the application into Kubernetes
+
+Once you have the Kubernetes cluster and Temporal is installed, it's time to install the application. We have created some manifests that will take care of installing the application for you. As is best practise, the OMS application will live in it's own Kubernetes namespace "oms". To ensure this namespace is present before we try and install anything into it, create the namespace using `kubectl`:
+
+```sh
+kubectl apply -f ./deployments/oms-namespace.yaml
+```
+
+Once that has completed, you can then install all of the application:
+
+```sh
+kubectl apply -f ./deployments/
+```
+
+To ensure the application is running you can check on the status of pods:
+
+```sh
+kubectl get pods -n oms
+```
+
+You should see something like the following:
+
+```
+NAME                              READY   STATUS    RESTARTS   AGE
+billing-api-0                     1/1     Running   0          24h
+billing-worker-5498c6ffd4-klfwk   1/1     Running   0          24h
+codec-server-847b59b84f-9ct54     1/1     Running   0          24h
+main-api-0                        1/1     Running   0          24h
+main-worker-777c59ccdb-jx46w      1/1     Running   0          24h
+web-557c8dc97-p5fkq               1/1     Running   0          24h
+```
+
+Once the READY column shows "1/1" for all the pods, the application is up.
+
+Please note that while in most regards the deployment is a standard Kubernetes setup, there is one thing which is a little unusual. Our API services make use of a cache to ensure that they can serve the listings for orders and shipments quickly. This cache lives in an SQLite file on disk, which makes the API services stateful. If an API pod needs to be restarted (due to upgrade, node failure etc) then it must be brought back up with the same disk as before to ensure it does not lose it's cache. For this reason, the API services use Statefulset instead of Deployment. The cache is mounted via a Kubernetes volume to ensure that the pod will maintain it's disk between runs.
```

Comment:
```suggestion
Please note that while in most regards the deployment is a standard Kubernetes setup, there is one thing which is a little unusual. Our API services make use of a cache to ensure that they can serve the listings for orders and shipments quickly. This cache lives in an SQLite file on disk, which makes the API services stateful. If an API pod needs to be restarted (due to upgrade, node failure etc) then it must be brought back up with the same disk as before to ensure it does not lose its cache. For this reason, the API services use Statefulset instead of Deployment. The cache is mounted via a Kubernetes volume to ensure that the pod will maintain it's disk between runs.
```